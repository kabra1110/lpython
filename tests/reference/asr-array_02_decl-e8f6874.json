{
    "basename": "asr-array_02_decl-e8f6874",
    "cmd": "lpython --show-asr --no-color {infile} -o {outfile}",
    "infile": "tests/../integration_tests/array_02_decl.py",
    "infile_hash": "8daa77dd2d5fe6c6f5f3ce867746c5e13290305ef7e1723ac9669285",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "asr-array_02_decl-e8f6874.stdout",
    "stdout_hash": "161c1c315c83399ca9f2f303efe354535ecf820e5529dcddc5b9a059",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}