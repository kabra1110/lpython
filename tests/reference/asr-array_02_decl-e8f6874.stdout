(TranslationUnit
    (SymbolTable
        1
        {
            _global_symbols:
                (Module
                    (SymbolTable
                        206
                        {
                            _lpython_main_program:
                                (Function
                                    (SymbolTable
                                        205
                                        {
                                            
                                        })
                                    _lpython_main_program
                                    (FunctionType
                                        []
                                        ()
                                        Source
                                        Implementation
                                        ()
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        []
                                        []
                                        .false.
                                    )
                                    [declare_arrays]
                                    []
                                    [(SubroutineCall
                                        206 declare_arrays
                                        ()
                                        []
                                        ()
                                    )]
                                    ()
                                    Public
                                    .false.
                                    .false.
                                    ()
                                ),
                            accept_multidim_f32_array:
                                (Function
                                    (SymbolTable
                                        201
                                        {
                                            _lpython_return_variable:
                                                (Variable
                                                    201
                                                    _lpython_return_variable
                                                    []
                                                    ReturnVar
                                                    ()
                                                    ()
                                                    Default
                                                    (Real 4)
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                ),
                                            xf32:
                                                (Variable
                                                    201
                                                    xf32
                                                    []
                                                    InOut
                                                    ()
                                                    ()
                                                    Default
                                                    (Array
                                                        (Real 4)
                                                        [(()
                                                        ())]
                                                        DescriptorArray
                                                    )
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                )
                                        })
                                    accept_multidim_f32_array
                                    (FunctionType
                                        [(Array
                                            (Real 4)
                                            [(()
                                            ())]
                                            DescriptorArray
                                        )]
                                        (Real 4)
                                        Source
                                        Implementation
                                        ()
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        []
                                        []
                                        .false.
                                    )
                                    []
                                    [(Var 201 xf32)]
                                    [(=
                                        (Var 201 _lpython_return_variable)
                                        (ArrayItem
                                            (Var 201 xf32)
                                            [(()
                                            (IntegerConstant 0 (Integer 4))
                                            ())]
                                            (Real 4)
                                            RowMajor
                                            ()
                                        )
                                        ()
                                    )
                                    (Return)]
                                    (Var 201 _lpython_return_variable)
                                    Public
                                    .false.
                                    .false.
                                    ()
                                ),
                            accept_multidim_f64_array:
                                (Function
                                    (SymbolTable
                                        202
                                        {
                                            _lpython_return_variable:
                                                (Variable
                                                    202
                                                    _lpython_return_variable
                                                    []
                                                    ReturnVar
                                                    ()
                                                    ()
                                                    Default
                                                    (Real 8)
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                ),
                                            xf64:
                                                (Variable
                                                    202
                                                    xf64
                                                    []
                                                    InOut
                                                    ()
                                                    ()
                                                    Default
                                                    (Array
                                                        (Real 8)
                                                        [(()
                                                        ())
                                                        (()
                                                        ())]
                                                        DescriptorArray
                                                    )
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                )
                                        })
                                    accept_multidim_f64_array
                                    (FunctionType
                                        [(Array
                                            (Real 8)
                                            [(()
                                            ())
                                            (()
                                            ())]
                                            DescriptorArray
                                        )]
                                        (Real 8)
                                        Source
                                        Implementation
                                        ()
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        []
                                        []
                                        .false.
                                    )
                                    []
                                    [(Var 202 xf64)]
                                    [(=
                                        (Var 202 _lpython_return_variable)
                                        (ArrayItem
                                            (Var 202 xf64)
                                            [(()
                                            (IntegerConstant 0 (Integer 4))
                                            ())
                                            (()
                                            (IntegerConstant 1 (Integer 4))
                                            ())]
                                            (Real 8)
                                            RowMajor
                                            ()
                                        )
                                        ()
                                    )
                                    (Return)]
                                    (Var 202 _lpython_return_variable)
                                    Public
                                    .false.
                                    .false.
                                    ()
                                ),
                            accept_multidim_i32_array:
                                (Function
                                    (SymbolTable
                                        199
                                        {
                                            _lpython_return_variable:
                                                (Variable
                                                    199
                                                    _lpython_return_variable
                                                    []
                                                    ReturnVar
                                                    ()
                                                    ()
                                                    Default
                                                    (Integer 4)
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                ),
                                            xi32:
                                                (Variable
                                                    199
                                                    xi32
                                                    []
                                                    InOut
                                                    ()
                                                    ()
                                                    Default
                                                    (Array
                                                        (Integer 4)
                                                        [(()
                                                        ())
                                                        (()
                                                        ())]
                                                        DescriptorArray
                                                    )
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                )
                                        })
                                    accept_multidim_i32_array
                                    (FunctionType
                                        [(Array
                                            (Integer 4)
                                            [(()
                                            ())
                                            (()
                                            ())]
                                            DescriptorArray
                                        )]
                                        (Integer 4)
                                        Source
                                        Implementation
                                        ()
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        []
                                        []
                                        .false.
                                    )
                                    []
                                    [(Var 199 xi32)]
                                    [(=
                                        (Var 199 _lpython_return_variable)
                                        (ArrayItem
                                            (Var 199 xi32)
                                            [(()
                                            (IntegerConstant 0 (Integer 4))
                                            ())
                                            (()
                                            (IntegerConstant 0 (Integer 4))
                                            ())]
                                            (Integer 4)
                                            RowMajor
                                            ()
                                        )
                                        ()
                                    )
                                    (Return)]
                                    (Var 199 _lpython_return_variable)
                                    Public
                                    .false.
                                    .false.
                                    ()
                                ),
                            accept_multidim_i64_array:
                                (Function
                                    (SymbolTable
                                        200
                                        {
                                            _lpython_return_variable:
                                                (Variable
                                                    200
                                                    _lpython_return_variable
                                                    []
                                                    ReturnVar
                                                    ()
                                                    ()
                                                    Default
                                                    (Integer 8)
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                ),
                                            xi64:
                                                (Variable
                                                    200
                                                    xi64
                                                    []
                                                    InOut
                                                    ()
                                                    ()
                                                    Default
                                                    (Array
                                                        (Integer 8)
                                                        [(()
                                                        ())
                                                        (()
                                                        ())
                                                        (()
                                                        ())]
                                                        DescriptorArray
                                                    )
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                )
                                        })
                                    accept_multidim_i64_array
                                    (FunctionType
                                        [(Array
                                            (Integer 8)
                                            [(()
                                            ())
                                            (()
                                            ())
                                            (()
                                            ())]
                                            DescriptorArray
                                        )]
                                        (Integer 8)
                                        Source
                                        Implementation
                                        ()
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        []
                                        []
                                        .false.
                                    )
                                    []
                                    [(Var 200 xi64)]
                                    [(=
                                        (Var 200 _lpython_return_variable)
                                        (ArrayItem
                                            (Var 200 xi64)
                                            [(()
                                            (IntegerConstant 9 (Integer 4))
                                            ())
                                            (()
                                            (IntegerConstant 9 (Integer 4))
                                            ())
                                            (()
                                            (IntegerConstant 9 (Integer 4))
                                            ())]
                                            (Integer 8)
                                            RowMajor
                                            ()
                                        )
                                        ()
                                    )
                                    (Return)]
                                    (Var 200 _lpython_return_variable)
                                    Public
                                    .false.
                                    .false.
                                    ()
                                ),
                            declare_arrays:
                                (Function
                                    (SymbolTable
                                        203
                                        {
                                            ac32:
                                                (Variable
                                                    203
                                                    ac32
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Array
                                                        (Complex 4)
                                                        [((IntegerConstant 0 (Integer 4))
                                                        (IntegerConstant 3 (Integer 4)))
                                                        ((IntegerConstant 0 (Integer 4))
                                                        (IntegerConstant 5 (Integer 4)))
                                                        ((IntegerConstant 0 (Integer 4))
                                                        (IntegerConstant 99 (Integer 4)))]
                                                        FixedSizeArray
                                                    )
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                ),
                                            ac64:
                                                (Variable
                                                    203
                                                    ac64
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Array
                                                        (Complex 8)
                                                        [((IntegerConstant 0 (Integer 4))
                                                        (IntegerConstant 10 (Integer 4)))
                                                        ((IntegerConstant 0 (Integer 4))
                                                        (IntegerConstant 13 (Integer 4)))
                                                        ((IntegerConstant 0 (Integer 4))
                                                        (IntegerConstant 11 (Integer 4)))
                                                        ((IntegerConstant 0 (Integer 4))
                                                        (IntegerConstant 16 (Integer 4)))]
                                                        FixedSizeArray
                                                    )
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                ),
                                            af32:
                                                (Variable
                                                    203
                                                    af32
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Array
                                                        (Real 4)
                                                        [((IntegerConstant 0 (Integer 4))
                                                        (IntegerConstant 3 (Integer 4)))]
                                                        FixedSizeArray
                                                    )
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                ),
                                            af64:
                                                (Variable
                                                    203
                                                    af64
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Array
                                                        (Real 8)
                                                        [((IntegerConstant 0 (Integer 4))
                                                        (IntegerConstant 10 (Integer 4)))
                                                        ((IntegerConstant 0 (Integer 4))
                                                        (IntegerConstant 4 (Integer 4)))]
                                                        FixedSizeArray
                                                    )
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                ),
                                            ai32:
                                                (Variable
                                                    203
                                                    ai32
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Array
                                                        (Integer 4)
                                                        [((IntegerConstant 0 (Integer 4))
                                                        (IntegerConstant 3 (Integer 4)))
                                                        ((IntegerConstant 0 (Integer 4))
                                                        (IntegerConstant 3 (Integer 4)))]
                                                        FixedSizeArray
                                                    )
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                ),
                                            ai64:
                                                (Variable
                                                    203
                                                    ai64
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Array
                                                        (Integer 8)
                                                        [((IntegerConstant 0 (Integer 4))
                                                        (IntegerConstant 10 (Integer 4)))
                                                        ((IntegerConstant 0 (Integer 4))
                                                        (IntegerConstant 10 (Integer 4)))
                                                        ((IntegerConstant 0 (Integer 4))
                                                        (IntegerConstant 10 (Integer 4)))]
                                                        FixedSizeArray
                                                    )
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                )
                                        })
                                    declare_arrays
                                    (FunctionType
                                        []
                                        ()
                                        Source
                                        Implementation
                                        ()
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        []
                                        []
                                        .false.
                                    )
                                    [accept_multidim_i32_array
                                    accept_multidim_i64_array
                                    accept_multidim_f32_array
                                    accept_multidim_f64_array]
                                    []
                                    [(Print
                                        ()
                                        [(FunctionCall
                                            206 accept_multidim_i32_array
                                            ()
                                            [((ArrayPhysicalCast
                                                (Var 203 ai32)
                                                FixedSizeArray
                                                DescriptorArray
                                                (Array
                                                    (Integer 4)
                                                    [((IntegerConstant 0 (Integer 4))
                                                    (IntegerConstant 3 (Integer 4)))
                                                    ((IntegerConstant 0 (Integer 4))
                                                    (IntegerConstant 3 (Integer 4)))]
                                                    DescriptorArray
                                                )
                                                ()
                                            ))]
                                            (Integer 4)
                                            ()
                                            ()
                                        )]
                                        ()
                                        ()
                                    )
                                    (Print
                                        ()
                                        [(FunctionCall
                                            206 accept_multidim_i64_array
                                            ()
                                            [((ArrayPhysicalCast
                                                (Var 203 ai64)
                                                FixedSizeArray
                                                DescriptorArray
                                                (Array
                                                    (Integer 8)
                                                    [((IntegerConstant 0 (Integer 4))
                                                    (IntegerConstant 10 (Integer 4)))
                                                    ((IntegerConstant 0 (Integer 4))
                                                    (IntegerConstant 10 (Integer 4)))
                                                    ((IntegerConstant 0 (Integer 4))
                                                    (IntegerConstant 10 (Integer 4)))]
                                                    DescriptorArray
                                                )
                                                ()
                                            ))]
                                            (Integer 8)
                                            ()
                                            ()
                                        )]
                                        ()
                                        ()
                                    )
                                    (Print
                                        ()
                                        [(FunctionCall
                                            206 accept_multidim_f32_array
                                            ()
                                            [((ArrayPhysicalCast
                                                (Var 203 af32)
                                                FixedSizeArray
                                                DescriptorArray
                                                (Array
                                                    (Real 4)
                                                    [((IntegerConstant 0 (Integer 4))
                                                    (IntegerConstant 3 (Integer 4)))]
                                                    DescriptorArray
                                                )
                                                ()
                                            ))]
                                            (Real 4)
                                            ()
                                            ()
                                        )]
                                        ()
                                        ()
                                    )
                                    (Print
                                        ()
                                        [(FunctionCall
                                            206 accept_multidim_f64_array
                                            ()
                                            [((ArrayPhysicalCast
                                                (Var 203 af64)
                                                FixedSizeArray
                                                DescriptorArray
                                                (Array
                                                    (Real 8)
                                                    [((IntegerConstant 0 (Integer 4))
                                                    (IntegerConstant 10 (Integer 4)))
                                                    ((IntegerConstant 0 (Integer 4))
                                                    (IntegerConstant 4 (Integer 4)))]
                                                    DescriptorArray
                                                )
                                                ()
                                            ))]
                                            (Real 8)
                                            ()
                                            ()
                                        )]
                                        ()
                                        ()
                                    )]
                                    ()
                                    Public
                                    .false.
                                    .false.
                                    ()
                                )
                        })
                    _global_symbols
                    []
                    .false.
                    .false.
                ),
            lpython_builtin:
                (IntrinsicModule lpython_builtin),
            main_program:
                (Program
                    (SymbolTable
                        204
                        {
                            _lpython_main_program:
                                (ExternalSymbol
                                    204
                                    _lpython_main_program
                                    206 _lpython_main_program
                                    _global_symbols
                                    []
                                    _lpython_main_program
                                    Public
                                )
                        })
                    main_program
                    [_global_symbols]
                    [(SubroutineCall
                        204 _lpython_main_program
                        ()
                        []
                        ()
                    )]
                ),
            numpy:
                (Module numpy)
        })
    []
)
